[EUWE] Restructured the Deployment Planning and Managing Infrastructure and …

suyogsainkar opened the pull request:
…Inventory guide and fixed the dead links to other titles. Also, moved the example postgresql config file at the end as an Appendix.

@dayleparker Hi Dayle,

I've made the following changes in the Deployment Planning and Managing Infrastructure and Inventory guide. Could you please peer review the updates and let me know if you have any feedback, or any other suggestions?

Deployment Planning Guide:

-Moved the following sections from Deployment Planning to a newly created section in the Managing Infrastructure and Inventory guide called Data Optimization Features. As that content isn't limited to a specific entity such as a Cluster, Host, or VM etc. (the way the Managing Infrastructure & Inventory is currently structured) but applies to all collected data, and discusses the optimization and utilization trend reports - all broadly classified from under the Optimize tab in the CFME UI - I therefore think that content requires its own section in the MI&I guide and does not fit in any of the existing sections. What do you think?

* 3.9. Optimization
* 3.10. Utilization Trends subsection
* 3.12. Bottlenecks
 
- Did not move chapter 3.8. Chart Features to the MI&I guide as it didn't seem to fit well in the newly created chapter (discussed above), nor in any other sections in the guide -- the chapter mainly talks about zooming and drilling into chart data in relation to the immediate C&U chart sections above it, so I think it's better left in its current place.

- As discussed, due to its odd placement in the middle of other sections, made section 2.4. Example PostgreSQL Configuration File into an Appendix at the end of the guide, now residing in a new file in the common directory (finally, eh :).

- Updated any references to the above sections since moved to a new place.

- Fixed several broken links (in Capacity Planning) to sections in other CF titles.  

Managing Infrastructure and Inventory:

- Created new chapter 6. Data Optimization Features and moved the following sections from Deployment Planning under it:

6.1. Optimization
6.1.1. Viewing Utilization Trend Summary
6.1.2. Viewing Detail Lines of a Utilization Trend
6.1.3. Viewing a Report of a Utilization Trend
6.2. Bottlenecks
6.2.1. Prerequisites
6.2.2. Viewing the Bottleneck Summary
6.2.3. Viewing a Report of the Bottlenecks Trend

- Linked the above chapter from the Viewing Capacity and Utilization Charts section for Clusters, Hosts, VM, and Datastores.

- Moved related image/icon files to MI&I image folder

Preview: http://file.bne.redhat.com/ssainkar/Red_Hat_CloudForms-4.2-Deployment_Planning_Guide-en-US.pdf

Preview: http://file.bne.redhat.com/ssainkar/Red_Hat_CloudForms-4.2-Managing_Infrastructure_and_Inventory-en-US.pdf

PR (Euwe): https://github.com/ManageIQ/manageiq_docs/pull/350

By the way, I had first created the following WIP PR for the Fine branch which got merged manually to master in an attempt to resolve merge conflict with subsequent updates. I will make changes to it, if required, based on any peer review feedback.  

PR (Fine): https://github.com/ManageIQ/manageiq_docs/pull/325
 
Please let me know if you have any questions.

Many thanks in advance!
Suyog

dayleparker commented:
@suyogsainkar 
Hi Suyog,
Thanks for all your work on this big reshuffle. It looks good! I have some feedback and spotted a few minor fixes, but it looks mostly ready. Please let me know if you have any questions or want to chat about it. I didn't test the links in this BTW since they're pointing to new uncreated guides.. hope that's cool. The same feedback will apply to euwe I imagine as well.

Cheers,
Dayle

-------------
regarding:
    3.9. Optimization
    3.10. Utilization Trends subsection
    3.12. Bottlenecks

-- I agree, that content didn't really fit in Deployment Planning in the way we think about the guide.
-- One question I had while reading was - do you have to set up Capacity and Utilization to use the Optimization features?
-- I imagine this (below) is legacy text, but I spotted a few grammatical issues in these sections. Do you want me to create a new bug, or do you want to fix the following here?

In 6.1. OPTIMIZATION
- In the Note - needs a full stop at the end (and the last sentence needs a bit of punctuation/some more concise rewording to make it read better).

In 6.1.1. Viewing Utilization Trend Summary
--> This is missing a word, should be "Viewing a Utilization Trend Summary" 
--> Same fix needed for "This procedure shows you how to view utilization trend summary" ==> should be "This procedure shows you how to view a utilization trend summary"

-----------
    Did not move chapter 3.8. Chart Features to the MI&I guide as it didn't seem to fit well in the newly created chapter (discussed above), nor in any other sections in the guide -- the chapter mainly talks about zooming and drilling into chart data in relation to the immediate C&U chart sections above it, so I think it's better left in its current place.

--> Makes sense to me. Looking at the section, there are a few grammar issues...

-------------
* In "3.8. CAPACITY AND UTILIZATION CHART FEATURES", the intro sentence reads oddly. Instead of "Capacity and utilization charts for host, clusters, virtual machines, and datastore provides its own set of special features including zooming in on a chart and shortcut menus.", could we update is similar to: 

"You can navigate Capacity and Utilization charts for hosts, clusters, virtual machines, and datastores by zooming in on a chart and using shortcut menus."

---------

    As discussed, due to its odd placement in the middle of other sections, made section 2.4. Example PostgreSQL Configuration File into an Appendix at the end of the guide, now residing in a new file in the common directory (finally, eh :).

-- Woo, nice one! :D

-------------------

Managing Infrastructure and Inventory:

_Comments:_
-- Overall - should "Capacity and Utilization" be capitalized always? (Like "Capacity and Utilization charts"?) I can see both in the two guides, but most of the time it's capitalized. Would be worth a quick run-through of both docs to make it consistent. (See 6.2. Bottlenecks, for example). But then 3.1 Capacity and Utilization Collection has "capacity and utilization data"... maybe when it's a CloudForms thing, like a chart or a role, it's capitals, and otherwise not? I'm not sure. Thoughts?

-----------

-- Last thought about the structure of "3.1 Capacity and Utilization Collection"... minor issue, but the order still seems a little mixed up.

* 3.4. ADDING DATABASE CREDENTIALS FOR DATA COLLECTION looks like it is more general, and I'd recommend moving this to before the provider-specific instructions.
* 3.7. DATA COLLECTED - I would recommend renaming this as it's a vague title. It looks like it's about the C&U charts themselves. How about renaming it to "Data Collected in Capacity and Utilization Charts"? (Or something better). I'd also recommend folding in "3.8. CAPACITY AND UTILIZATION CHART FEATURES" as a subsection, called something like "Navigating Capacity and Utilization Charts"

This would look like:

CHAPTER 3. CAPACITY PLANNING
3.1. CAPACITY AND UTILIZATION COLLECTION
3.2. ASSIGNING THE CAPACITY AND UTILIZATION SERVER ROLES
3.3. ADDING DATABASE CREDENTIALS FOR DATA COLLECTION  --- moved up
3.4. DATA COLLECTION FOR RED HAT ENTERPRISE VIRTUALIZATION --- keep the provider-specific stuff together....
3.5. DATA COLLECTION FOR RED HAT ENTERPRISE LINUX OPENSTACK PLATFORM
3.6. DATA COLLECTION FOR GOOGLE COMPUTE ENGINE
3.7. CAPACITY AND UTILIZATION CHARTS  --- revised title
3.7.1. DATA COLLECTED BY CAPACITY AND UTILIZATION CHARTS -- perhaps moved into a subsection?
3.7.2. NAVIGATING CAPACITY AND UTILIZATION CHARTS(was: CAPACITY AND UTILIZATION CHART FEATURES)
[3.9. OPTIMIZATION  -- remove as it's a duplicate]
3.9. UTILIZATION TRENDS -- is this needed now that the content is in MII? Maybe we could link to MII from the end of 3.7.2 and maybe 3.10?
3.10. PLANNING WHERE TO PUT A NEW VIRTUAL MACHINE

Note: folding together the sections into 3.7 is optional (what are your thoughts?). I'm also unsure if 3.9 (that's a duplicate of what's in MII now I am pretty sure) and 3.10 are needed here still.

suyogsainkar commented:
@dayleparker Hi Dayle, many thanks for your review and suggestions. I have made modifications and reworded some text where appropriate, see my responses below.

-"One question I had while reading was - do you have to set up Capacity and Utilization to use the Optimization features?"
Yes, optimization features work on the collected C&U data. 

-In "3.8. CAPACITY AND UTILIZATION CHART FEATURES", the intro sentence reads oddly."
You can have independent charts for host, clusters, VMs.. and zooming into a chart and drilling into chart data to see more details are some of the features associated with the chart.. is what the section/intro talks about. I think changing it to the suggested text would change its intended meaning and function.

-Previously, we agreed on having capacity and utilization capitalized only when used in a proper noun, such as when referring to server roles - Capacity and Utilization Coordinator, Capacity & Utilization Data Collector, or in titles or calling out a UI element. All other references to it in a sentence should not be capitalized. I had a quick look again in the Deployment Planning Guide and seems consistent with that rule. I had a quick look in the MI&I guide and fixed a couple of references to make it consistent. 

-I've removed the duplicate section 3.9. Optimization (now resides in MII), also moved 3.10 Utilization Trends to MII as it seemed a bit out of place in Deployment Planning since all related content now resides in MII.

-Adding a new title called 3.7. CAPACITY AND UTILIZATION CHARTS sounds like a good idea, however, that also means adding an extra (blank) level and a fourth level to sections 3.7.1. Capacity and Utilization Charts for Host, Clusters, and Virtual Machines and  3.7.2. Capacity and Utilization Charts for Datastores which I think is best avoided - so prefer keeping that as it is.

Please let me know what you think of the changes, otherwise it's ready for merging. By the way, I will revisit (and update if required) the links as soon as some of the newly added/modified sections have been merged (can only guess what the links to those sections will be).

dayleparker commented on doc-Managing_Infrastructure_and_Inventory/topics/Data_Optimization_Features.adoc:
```diff
@@ -0,0 +1,158 @@
+utilization trends, and identify and project bottlenecks in your
+environment. In addition, you can predict where you have capacity for
+additional virtual machines.
+
```
I recommend noting here that you need C&U enabled so that customers aren't confused (this is only because the content is now in a separate guide from the 'how to enable C&U' content.

dayleparker commented on doc-Deployment_Planning_Guide/topics/capacity.adoc:
```diff
@@ -344,14 +342,13 @@ Charts created include:
 
-Each chart provides its own set of special features including zooming in
-on a chart and shortcut menus.
+Capacity and utilization charts for host, clusters, virtual machines, and datastore provides its own set of special features including zooming in on a chart and shortcut menus.
```
Typos: This needs to be fixed to "hosts" and "datastores" .

dayleparker commented:
Hi @suyogsainkar, 
Thanks for making those updates, and considering the suggestions.
The updates look good, and I agree it makes a lot of sense to also move the Utilization Trends section as you have.

There are two last changes that are needed before we can merge this -- I've pointed them out in comments in the specific spots so it's clear (sorry if these comments were unclear before). Then it should be ready to merge.

Thanks again!
Dayle

suyogsainkar commented:
Thanks @dayleparker. I've fixed those typos and updated the note to mention the requirement for enabling C&U data for optimization features.

Additionally, I've also remove the Optimization subsection as it seemed redundant/duplicate, and just kept the main intro called Data Optimization (removed the word Feature) - everything under that falls under the Optimize tab/concept anyway, and there's no specific action called Optimization but it discusses Utilization trend, Bottleneck reports etc, and linked it from/to Planning Where to Put a New Virtual Machine as it's all related. 

Preview: http://file.bne.redhat.com/ssainkar/Red_Hat_CloudForms-4.2-Managing_Infrastructure_and_Inventory-en-US-updated.pdf

http://file.bne.redhat.com/ssainkar/Red_Hat_CloudForms-4.2-Deployment_Planning_Guide-en-US-updated.pdf

Thanks!

dayleparker commented:
Hi Suyog,
Thanks so much for making those updates. Both books look awesome -- and very good call about the Optimization section! All looks good so merging. Please feel free to send over a PR for master anytime when the updates are there too. Thanks again for all your work! :)

suyogsainkar commented:
Thanks, Dayle!

suyogsainkar commented:
@dayleparker The cherrypicking didn't go as planned due to several changes in multiple files, so made the additional updates in master (fine) branch manually via https://github.com/ManageIQ/manageiq_docs/pull/381 as suggested by Andrew. Also updated the memory requirement to 12 GB under 1.2. Requirements 1.2.1. Virtual Hardware Requirements.

Preview:
http://file.bne.redhat.com/ssainkar/Red_Hat_CloudForms-4.5-Deployment_Planning_Guide-en-US.pdf
http://file.bne.redhat.com/ssainkar/Red_Hat_CloudForms-4.5-Managing_Infrastructure_and_Inventory-en-US.pdf

Thanks!

dayleparker commented:
It all looks great to me, Suyog :) Nice work!
